File: ios/JJMproject/Classes/JSBridge/Modules/Polling/PollingModule.m
//
//  NetworkModule.m
//  MarioFramework
//
//  Created by MeePwn on 2017/11/15.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(PollingModule, NSObject)

RCT_EXTERN_METHOD(startPolling:(NSString *)pollingID timeInterval:(double)timeInterval)
RCT_EXTERN_METHOD(stopPolling:(NSString *)pollingID resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/JJMproject/Classes/JSBridge/Modules/Polling/PollingModule.swift
//
//  PollingModule.swift
//  MarioFramework
//
//  Created by MeePwn on 2017/11/15.
//  Copyright © 2017年 Facebook. All rights reserved.
//

import UIKit

@objc(PollingModule)
class PollingModule: NSObject {

  private lazy var pollingUtility: PollingUtility = {
    return PollingUtility()
  }()
  
  @objc func startPolling(_ pollingID: String, timeInterval: Double) {
    pollingUtility.startPolling(pollingID: pollingID, timeInterval: timeInterval) { _ in
      // TODO 发送轮询成功信息
      print("== pollingID ===>>>> \(pollingID)")
    }
  }
  
  @objc func stopPolling(_ pollingID: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    pollingUtility.stopPolling(pollingID: pollingID)
    resolve(pollingID)
  }
  
}
